autogen.sh

```diff
@@ -67,6 +67,138 @@ sync_external () {
     cp -pPR "$srcdir" "$destdir"
 }
 
+########################################################################
+## Step functions
+########################################################################
```

Comment:
All this is orthogonal to the goal of this PR. I'd rather consider it separately.

Reply:
Sure. I only realized later that the original `--do=xxx` is to run a particular step rather than to turn on/off a particular option. At that point, I need to switch to `--with-xxx` as options to switch on/off features. That's why many refactoring mixed in. I'll see if I can split them. Do you prefer the refactoring of `autogen.sh` goes before this PR or after? This PR needs finer options to skip module initializations, and it will be significantly cleaner if it's after the refactoring.

Reply:
Sure, let's put the refactoring first.

Reply:
Done. Search for `PREBUILT` to find the changes.